olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -33,6 +33,11 @@
 #include "repositories/PartitionsRepository.h"
 #include "repositories/PrefetchTilesRepository.h"
 
+// Needed to avoid endless warnings from GetVersion/WithVersion
+#include <olp/core/porting/warning_disable.h>
+PORTING_PUSH_WARNINGS()
+PORTING_CLANG_GCC_DISABLE_WARNING("-Wdeprecated-declarations")
+
```

Comment:
We should rather avoid calling GetVersion, and the warning will not be triggered.

Reply:
Yes, definitely, but this is a bigger change and is not so easy. We basically need to change all internal classes to not use the version from the request structures anymore. This needs to be done once we remove the deprecated accessors from the request structures.